core/lib/contracts/src/lib.rs

```diff
@@ -288,12 +292,50 @@ pub fn read_zbin_bytecode(relative_zbin_path: impl AsRef<Path>) -> Vec<u8> {
     read_zbin_bytecode_from_path(bytecode_path)
 }
 
+pub fn read_yul_bytecode(relative_artifacts_path: PathBuf, name: &str) -> Vec<u8> {
+    let artifacts_path = Path::new(&home_path()).join(relative_artifacts_path);
+    read_yul_bytecode_by_path(artifacts_path, name)
+}
+
+pub fn read_yul_bytecode_by_path(artifacts_path: PathBuf, name: &str) -> Vec<u8> {
+    let bytecode_path = artifacts_path.join(format!("{0}.yul/{0}.yul.zbin", name));
+
+    if fs::exists(&bytecode_path).unwrap_or_else(|_| panic!("Invalid path: {:?}", &bytecode_path)) {
```

Comment:
Nit: I'd add the error to the panic message just in case. Also, `bytecode_path` can be interpolated in the panic string; ditto for panics below.